eth/client.go

```diff
@@ -298,7 +298,45 @@ func (cl *Client) SubscribeToDeposited(ctx context.Context, contract *bindings.E
 	if err != nil {
 		return err
 	}
-	it, err := contract.FilterDeposited(fOpts, epochs, accs)
+	it, err := contract.FilterExiting(fOpts, epochs, accs)
+	defer it.Close()
+	if err != nil {
+		return err
+	}
+	for it.Next() {
+		sink <- it.Event
+	}
+	// Wait for error or ctx done.
+	select {
+	case err := <-sub.Err():
+		return err
+	case <-ctx.Done():
+		return ctx.Err()
+	}
```

Comment:
You mean `<-ctx.Done()` should not be part of the select?  
We also do it like this in `go-perun` since i think that the go-ethereum subscription does not propagate up the correct context-cancelled error, but im not sure.

Reply:
I can also change the subscribe functions to return a sub type, but i wrote them in parallel to yours, so they are designed differently :smile:

Reply:
I'm pretty sure I didn't write the other sub function. You can use `git blame` if you're interested ;)

Sure that we did it like this in `go-perun`? E.g. for `Adjudicator.Subscribe` we have our own subscription type which has an iterator API.

I think the API of these subscribe functions is a bit confusing, because you would assume the `SubscribeX` function to return after successfully setting up the sub. You could change it so that you create the channel internally and then return the channel and the `ethereum.Subscription`.

Reply:
I know that it is a bad API, was just because of time pressure. I can change the Sub functions to all return immediately, will just take some time to refactor then.  
  
[Here](https://git.perun.network/perun/go-perun/-/blob/dev/backend/ethereum/channel/withdraw.go#L54) for example we also wait on the ctx that created the watch opts.

Reply:
No idea who wrote the other sub functions :see_no_evil: thought it was you or Matthias.

Reply:
So i will add `FrozenSubscription` and `ExitingSubscription` types.